clients/src/main/java/org/apache/kafka/clients/consumer/internals/StreamsInitializeRequestManager.java

```diff
@@ -16,23 +16,106 @@
+        final List<StreamsInitializeRequestData.Subtopology> topology = new ArrayList<>(subTopologyMap.size());
+        for (final Map.Entry<String, StreamsAssignmentInterface.SubTopology> subtopology : subTopologyMap.entrySet()) {
+            topology.add(getSubtopologyFromStreams(subtopology.getKey(), subtopology.getValue()));
+        }
+        return topology;
+    }
+
+    private static StreamsInitializeRequestData.Subtopology getSubtopologyFromStreams(final String subtopologyName,
+                                                                                      final StreamsAssignmentInterface.SubTopology subtopology) {
+        final StreamsInitializeRequestData.Subtopology subtopologyData = new StreamsInitializeRequestData.Subtopology();
+        subtopologyData.setSubtopology(subtopologyName);
+        subtopologyData.setSourceTopics(new ArrayList<>(subtopology.sourceTopics));
+        subtopologyData.setSinkTopics(new ArrayList<>(subtopology.sinkTopics));
+        subtopologyData.setRepartitionSourceTopics(getRepartitionTopicsInfoFromStreams(subtopology));
+        subtopologyData.setStateChangelogTopics(getChangelogTopicsInfoFromStreams(subtopology));
+        return subtopologyData;
+    }
+
+    private static List<StreamsInitializeRequestData.TopicInfo> getRepartitionTopicsInfoFromStreams(final StreamsAssignmentInterface.SubTopology subtopologyDataFromStreams) {
+        final List<StreamsInitializeRequestData.TopicInfo> repartitionTopicsInfo = new ArrayList<>();
+        for (final Map.Entry<String, StreamsAssignmentInterface.TopicInfo> repartitionTopic : subtopologyDataFromStreams.repartitionSourceTopics.entrySet()) {
+            final StreamsInitializeRequestData.TopicInfo repartitionTopicInfo = new StreamsInitializeRequestData.TopicInfo();
+            repartitionTopicInfo.setName(repartitionTopic.getKey());
+            repartitionTopic.getValue().numPartitions.ifPresent(repartitionTopicInfo::setPartitions);
+            repartitionTopicsInfo.add(repartitionTopicInfo);
+        }
+        return repartitionTopicsInfo;
+    }
+
+    private static List<StreamsInitializeRequestData.TopicInfo> getChangelogTopicsInfoFromStreams(final StreamsAssignmentInterface.SubTopology subtopologyDataFromStreams) {
+        final List<StreamsInitializeRequestData.TopicInfo> changelogTopicsInfo = new ArrayList<>();
+        for (final Map.Entry<String, StreamsAssignmentInterface.TopicInfo> changelogTopic : subtopologyDataFromStreams.stateChangelogTopics.entrySet()) {
+            final StreamsInitializeRequestData.TopicInfo changelogTopicInfo = new StreamsInitializeRequestData.TopicInfo();
+            changelogTopicInfo.setName(changelogTopic.getKey());
+            changelogTopicsInfo.add(changelogTopicInfo);
+        }
+        return changelogTopicsInfo;
+    }
+    
+    private void onResponse(final ClientResponse response, final Throwable exception) {
+        if (exception != null) {
+            // todo: handle error
```

Comment:
maybe we want to log something for now so that we at least see what's going on during end-to-end testing? Same for the other branch